concurrency/_index.md

```diff
@@ -1,7 +1,158 @@
+    <artifactId>jakarta.enterprise.concurrent-api</artifactId>
+    <version>3.0.2</version>
+</dependency>
+```
+
+### Asynchronous Tasks
+
+Jakarta EE compliant containers must contain a `ManagedExecutorService`, which extends the Java SE `ExecutorService`. Tasks can be passed to a `ManagedExecutorService`, and those tasks can then be performed within a container managed thread and passed back to the caller once complete.
+This process can occur asynchronously, allowing the application to continue performing other tasks while the `ManagedExecutorService` processes the task which had been handed off.
+The `ManagedExecutorService` returns a `Future` object to the caller, which will return a `boolean` indicating when the task has been completed.
+The caller can poll this `Future` object and then perform some action using the resulting object once the task has completed.
+
+A task is developed by creating a class that must implement `java.util.concurrent.Callable` or `java.lang.Runnable`, or any of the functional interfaces that can be supplied to a `java.util.concurrent.CompletionStage`.
+A Callable task class can optionally return a result to the caller. When tasks are submitted to a managed executor or managed `CompletionStage`, they become contextual, behaving in the same manner as they had within the original container.
+This means that contextual information and state can be passed to the `ManagedExecutorService` within the task and utilized while processing.
+For example, a session may contain the username of an authenticated user, and that information can be passed to the `ManagedExecutorService` within the task. A task class that implements `Runnable` may be structured as follows:
+
+```Java
+public class MyTask implements Runnable {
+    // Inject resources, if needed
+
+    public void run() {
+        // execute logic
+    }
+}
+```
+
+For more fine grained control, a task class can optionally implement the `jakarta.enterprise.concurrent.ManagedTask` interface, which enables the ability to perform actions upon lifecycle events.
+This is achieved by registering a `jakarta.enterprise.concurrent.ManagedTaskListener` instance with the task. A use case for implementing `ManagedTask` may be to initiate a process when a particular lifecycle event occurs.
+
+To pass a task to a `ManagedExecutorService`, the `ManagedExecutorService` can be obtained for use within a resource through use of the `Java Naming and Directory Interface (JNDI)`.
+Alternatively, the `ManagedExecutorService` can be configured via XML within the `web.xml` of the application.
+After the resource has been obtained, then the task can be submitted to it by calling upon any one of the `submit()`, `execute()`, `invokeAll()`, `invokeAny()`, `runAsync()`, or `supplyAsync()` methods.
+The following code demonstrates how to obtain the `ManagedExecutorService` resource and submit a task to it.
+
+```Java
+@Resource(name = "concurrent/_defaultManagedExecutorService")
```

Comment:
The specification-defined default managed executor service has the following name:
```suggestion
@Resource(lookup = "java:comp/DefaultManagedExecutorService")
```
If this isn't what you were intending, choose a different name that is less likely to cause confusion.